lib/game.rb

```diff
@@ -0,0 +1,131 @@
+require 'pry'
+class Game
+
+  extend Players
+
+  attr_accessor :board, :player_1, :player_2
+
+  WIN_COMBINATIONS = [
+    [0, 1, 2],
+    [3, 4, 5],
+    [6, 7, 8],
+    [0, 3, 6],
+    [1, 4, 7],
+    [2, 5, 8],
+    [0, 4, 8],
+    [6, 4, 2]
+  ]
+
+  def initialize(player_1 = Players::Human.new("X"), player_2 = Players::Human.new("O"), board = Board.new)
+    @player_1 = player_1
+    @player_2 = player_2
+    @board = board
+  end
+
+  def current_player
+    self.board.turn_count % 2 == 0 ? self.player_1 : self.player_2
+  end
+
+  def over?
+    (self.draw? || self.won?) ? true : false
+  end
+
+  def won?
+    number_of_O = self.board.cells.select {|cell| cell == "O"}
+    number_of_X = self.board.cells.select {|cell| cell == "X"}
+    WIN_COMBINATIONS.detect do |win_combination|
+      win_index_1 = win_combination[0]
+      win_index_2 = win_combination[1]
+      win_index_3 = win_combination[2]
+
+      position_1 = self.board.cells[win_index_1]
+      position_2 = self.board.cells[win_index_2]
+      position_3 = self.board.cells[win_index_3]
+
+      if self.board.full?
+        position_1 == "X" && position_2 == "X" && position_3 == "X"
+      else
+        (position_1 == "X" && position_2 == "X" && position_3 == "X") || (position_1 == "O" && position_2 == "O" && position_3 == "O")
+      end
+    end
+  end
+
+  def draw?
+    if self.won? || !self.board.full?
+      false
+    else
+      true
+    end
+  end
+
+  def winner
+    if self.won? != nil
```

Comment:
Instead of calling `Game#won?` twice, consider using an if statement assignment. For example:

```ruby
if won = won?
  board.cells[won.first]
end
```

If `won?` returns true, then and only then will the local variable `won` be assigned, otherwise, the method will just abort with a false return value. But if `won` is assigned, we can use that variable instead of calling `won?` again within the if statement.